test/Coinbase/Exchange/Socket/Test.hs

```diff
@@ -97,3 +110,667 @@ waitCancelThreads action1 action2 = do
+        <*> arbitrary
+    , TickerMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    , L2SnapshotMsg <$> arbitrary <*> arbitrary <*> arbitrary
+    , L2UpdateMsg <$> arbitrary <*> arbitrary <*> arbitrary
+    , ReceivedLimitMsg <$> arbitrary  <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    , ReceivedMarketMsg <$> arbitrary  <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary
+    , OpenMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary
+    , MatchMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    , LastMatchMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    , DoneMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    , ChangeLimitMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    , ChangeMarketMsg <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+        <*> arbitrary <*> arbitrary <*> arbitrary
+    ]
+
+--------------------------------------------------------------------------------
+-- `Arbitrary` instances for types defined in `... .MarketData.Types`
+-- FIXME: These should be moved to the module in which these types are tested,
+-- but the types themselves should probably move to `... .Core.Types` or
+-- `... Core`.
+
+instance Arbitrary Size where
+  arbitrary = Size <$> arbitrary
+
+instance Arbitrary Price where
+  arbitrary = Price <$> arbitrary
+
+instance Arbitrary Cost where
+  arbitrary = Cost <$> arbitrary
+
+instance Arbitrary CurrencyId where
+  arbitrary = CurrencyId <$> arbitrary
+
+instance Arbitrary ProductId where
+  arbitrary = ProductId <$> arbitrary
+
+instance Arbitrary Side where
+  arbitrary = elements [Buy, Sell]
+
+instance Arbitrary CoinScientific where
+  arbitrary = CoinScientific <$> arbitrary
+
+instance Arbitrary Sequence where
+  arbitrary = Sequence <$> arbitrary
+
+instance Arbitrary TradeId where
+  arbitrary = TradeId <$> arbitrary
+
+instance Arbitrary OrderId where
+  arbitrary = OrderId <$> arbitrary
+
+instance Arbitrary ClientOrderId where
+  arbitrary = ClientOrderId <$> arbitrary
+
+instance Arbitrary Reason where
+  arbitrary = elements [Filled, Canceled]
+
+--------------------------------------------------------------------------------
+
+-- FIXME: This is a hack. Newer versions of the `quickcheck-instances` package
+-- provide a proper `Arbitrary` instance for `UUID`. Delete this after updating
+-- package dependencies.
+instance Arbitrary UUID where
+  arbitrary = elements [nil]
```

Comment:
This is too much extra code! Let's add any necessary dependencies and just derive these.

Reply:
Yes, it's quite absurd. I believe we can derive all or most of these using `generic-random`; I'll re-write using that.